schema/connector-discovery-schema.json

```diff
@@ -0,0 +1,260 @@
+{
+  "$schema": "http://json-schema.org/schema#",
+  "title": "Connector discovery schema",
+  "type": "object",
+  "properties": {
+    "image": {
+      "$ref": "#/definitions/link"
+    },
+    "test_auth": {
+      "$ref": "#/definitions/link"
+    },
+    "object_types": {
+      "$ref": "#/definitions/object_types"
+    },
+    "actions": {
+      "$ref": "#/definitions/actions"
+    },
+    "config": {
+      "$ref": "#/definitions/configs"
+    }
+  },
+  "definitions": {
+    "link": {
+      "type": "object",
+      "properties": {
+        "href": {
+          "type": "string",
+          "format": "uri"
+        }
+      },
+      "required": [
+        "href"
+      ]
+    },
+    "object_types": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/object_type"
+      }
+    },
+    "object_type": {
+      "type": "object",
+      "properties": {
+        "doc": {
+          "$ref": "#/definitions/link"
+        },
+        "fields": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/object_input_field"
+          }
+        },
+        "pollable": {
+          "type": "boolean"
+        },
+        "metadata": {
+          "type": "object"
+        },
+        "endpoint": {
+          "$ref": "#/definitions/link"
+        }
+      },
+      "required": [
+        "endpoint"
+      ]
+    },
+    "object_input_field": {
+      "type": "object",
+      "properties": {
+        "regex": {
+          "type": "string"
+        },
+        "capture_group": {
+          "type": "integer"
+        },
+        "env": {
+          "type": "string"
+        },
+        "required": {
+          "type": "boolean"
+        }
+      },
+      "dependencies": {
+        "regex": {
+          "required": [
+            "capture_group"
+          ]
+        }
+      }
+    },
+    "actions": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/connector_level_action"
+      }
+    },
+    "connector_level_action": {
+      "type": "object",
+      "properties": {
+        "url": {
+          "$ref": "#/definitions/link"
+        },
+        "user_input": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/connector_action_user_input"
+          }
+        },
+        "request": {
+          "type": "object",
+          "additionalProperties": {
+            "type": ["boolean", "string", "number", "object"],
```

Comment:
I'm not sure I'm handling the property "request", best way here.

What I know is that, it can be used to silently collect (without asking user) some ENV variables from client.
Also, connector authors probably want to specify some variables of their own. (For eg something like `"task_type": "incident"`)

Reply:
Removed the "request" property.